api/users/models.py

```diff
@@ -195,15 +195,22 @@ def available_actions(self):
             .annotate(missed_requirements=Count('requirements',
                                                 filter=~Q(requirements__in=self.requirements.all()))) \
             .filter(missed_requirements=0) \
+            .annotate(has_volunteered=Count('interested_volunteers',
+                                            filter=Q(interested_volunteers__id=self.id))) \
+            .filter(has_volunteered=0) \
             .filter(resident__ward__in=self.wards.all()) \
             .filter(help_type__in=self.help_types.all())
 
     @property
     def upcoming_actions(self):
```

Comment:
Shouldn't this list only show actions that are `ActionStatus.ASSIGNED` to this volunteer?

Reply:
Nope, that's the list of all available actions. This is why I had to add the filtering to only show the ones they haven't registered interest yet.

The `ActionStatus.ASSIGNED` are part of the `upcoming_actions`, along with the `ActionStatus.INTEREST` where the user is part of the volunteer. 


Actually that `upcoming_actions` method needs a little tweak to filter out the actions that are not with `ActionStatus.INTEREST` in the `union`. Otherwise users will see action they were interested in, but didn't get assigned to them and it'll just clutter their list. I'll take care of that tomorrow.

Reply:
Sorry, comment was regarding the `upcoming_actions` function - which seems to return actions that are ongoing, completed, and couldn't complete. I'd have thought it should only show those that are assigned, and those that you've specified interested in? :)

Reply:
That one actually filters out the `ONGOING`, `COMPLETED` and `COULDNT_COMPLETE` from the actions of the user, but you're right, it would read better if it was just including the ones that are `ASSIGNED`. Actually, if we made sure the person is listed in the interested volunteer when the action is assigned on the admin, it would just need to be the `self.actions_interested_in.all()`. I'll set that in place this morning.

Reply:
`exclude` 🤦

Reply:
Sorry mate, helps if I have eyes.

Reply:
The fact that you missed it meant the code wasn't super clear :) I've updated how `upcoming_actions` are computed and made sure the list of interested volunteers always contains the assigned volunteer. Had to wait for a transaction to be committed again, like when we create the notifications. A bit cumbersome...